### PR TITLE
Update some dependencies for compilation with pc-windows-gnu

### PR DESCRIPTION
**Pull request by tomaka**
Integrates https://github.com/paritytech/rust-rocksdb/pull/25, https://github.com/paritytech/rust-snappy/pull/9, https://github.com/paritytech/libusb-sys/pull/2 and https://github.com/paritytech/libusb-sys/pull/4

cc @TriplEight

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -2222,7 +2222,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 dependencies = [
  "num-integer 0.1.39 (registry+https://github.com/rust-lang/crates.io-index)",
  "num-traits 0.2.6 (registry+https://github.com/rust-lang/crates.io-index)",
- "rand 0.4.3 (registry+https://github.com/rust-lang/crates.io-index)",
+ "rand 0.3.22 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (ordian) [inline Cargo.lock]: This is probably downgrade from https://github.com/rust-lang/cargo/issues/5702, do we want to revert it?

Comment (tomaka) [reply Cargo.lock]: I'm surprised we're not using `cargo build --frozen` in our CI to detect that kind of problems.

**Comment by ordian** [comment]
Also android build is failing.

**Comment by tomaka** [comment]
> Also android build is failing.

That's quite funny. We used to erroneously build libusb-sys for x86-64-linux on the Android pipeline, but I assume that it worked because we don't actually use libusb-sys from within Parity when on Android.

Now that this is fixed, we reach a `#error "Unsupported OS"`.

**Comment by niklasad1** [comment]
Ok, shall we just not use the `hardware-wallet stuff` for android temporary?

**Comment by ordian** [comment]
I suggest to revert rand downgrade and deal with android in a separate PR. Any opinions?

File: Cargo.lock (hunk 1 above)
Comment (ordian) [reply Cargo.lock]: I guess `cargo update -p rand:0.3.22` should fix that. `cargo check --frozen` won't fail, the downgrade only happens if you do `cargo update -p <some dep>`.

**Comment by tomaka** [comment]
I don't think we should make `pc-windows-gnu` work by breaking Android.
I'll submit another PR.

**Pull request closed without merging**